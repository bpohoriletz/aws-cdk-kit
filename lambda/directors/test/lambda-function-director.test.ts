import * as cdk from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import LambdaFunctionDirector from '../lambda-function-director';
import GoLambdaFunctionBuilder from '../../lambda-function-builders/go-lambda-function-builder';

describe('new LambdaFunctionDirector()', () => {
  let instance: LambdaFunctionDirector;
  let stack: cdk.Stack;

  beforeEach(() => {
    stack = new cdk.Stack();
  });

  it('constructs role for EC2 lifecycle events', () => {
    instance = new LambdaFunctionDirector(GoLambdaFunctionBuilder);
    instance.constructSampleGoLambdaFunction(stack, '/go/main');

    expect(Template.fromStack(stack)).toMatchSnapshot();
  });
});
