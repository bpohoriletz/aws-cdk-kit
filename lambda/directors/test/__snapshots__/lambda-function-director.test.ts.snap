// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`new LambdaFunctionDirector() constructs role for EC2 lifecycle events 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "gomain66868564": {
      "DependsOn": [
        "gomainServiceRole67369D3C",
      ],
      "Properties": {
        "Code": {
          "S3Bucket": {
            "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
          },
          "S3Key": "05aa4bdf38c09631c3439d6ba1fb9556c0bfeb31415df93eda221af1b3688748.zip",
        },
        "Role": {
          "Fn::GetAtt": [
            "gomainServiceRole67369D3C",
            "Arn",
          ],
        },
        "Runtime": "provided.al2023",
      },
      "Type": "AWS::Lambda::Function",
    },
    "gomainServiceRole67369D3C": {
      "Properties": {
        "AssumeRolePolicyDocument": {
          "Statement": [
            {
              "Action": "sts:AssumeRole",
              "Effect": "Allow",
              "Principal": {
                "Service": "lambda.amazonaws.com",
              },
            },
          ],
          "Version": "2012-10-17",
        },
        "ManagedPolicyArns": [
          {
            "Fn::Join": [
              "",
              [
                "arn:",
                {
                  "Ref": "AWS::Partition",
                },
                ":iam::aws:policy/service-role/AWSLambdaBasicExecutionRole",
              ],
            ],
          },
        ],
      },
      "Type": "AWS::IAM::Role",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
